### PR TITLE
[Improvement-4884][Common] fix the subsequent processing of getting the YARN ResourceManager address to be empty

### LINKED ISSUE #4884: [Improvement][Common] a concatenation URL problem occurred when the address of the ACITVE ResourceManager in YARN HA mode failed to be obtained
**Describe the question**
Acitve ReSourceManager IP = null when obtaining Acitve ReSourceManager address in YARN HA mode fails,
the program will return splicing application url = http://null:8088/ws/v1/cluster/apps/%s 

**What are the current deficiencies and the benefits of improvement**
The yarn application URL generation failed exception should be thrown here

**Which version of DolphinScheduler:**
 -[dev]

### PR DESCRIPTION
**Pull request by zhuangchong**
## What is the purpose of the pull request
this PR close #4884 
Acitve ReSourceManager IP = null when obtaining Acitve ReSourceManager address in YARN HA mode fails,the yarn application URL generation failed exception should be thrown here

## Verify this pull request
This change added tests and can be verified as follows:
  - *Manually verified the change by testing locally.*

### CODE REVIEW

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/exception/BaseException.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.common.exception;
+
+/**
+ * Base Exception class for DolphinScheduler
+ */
+public class BaseException extends Exception {
```
Comment (CalvinKirs) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/exception/BaseException.java:23]: Custom exception naming should be related to the business

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/HadoopUtils.java [hunk 2]
```diff
@@ -664,12 +661,7 @@ public static String getAcitveRMName(String rmIds) {
                 }
 
             } catch (Exception e) {
-                for (int i = 1; i < rmIdArr.length; i++) {
-                    String  state = getRMState(String.format(yarnUrl, rmIdArr[i]));
-                    if (Constants.HADOOP_RM_STATE_ACTIVE.equals(state)) {
-                        return rmIdArr[i];
-                    }
```
Comment (CalvinKirs) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/HadoopUtils.java:671]: This area may need to be read by others. Although I think your approach is correct, we still need to consider historical reasons.

Comment (zhuangchong) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/HadoopUtils.java:671]: My understanding of this code is that the main code in the try code block is the HTTP request, and the exception is captured inside the HTTP request. I have a big question about capturing the exception here and conducting business processing this time, and I think it is not necessary.

---
我对此段代码的理解，try 代码块内主要代码是http请求，并且http请求内部捕获了异常，此次在此捕获异常并进行业务处理 我是存在很大疑问的，我认为是没有必要的。

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/exception/BaseException.java (hunk 1 above)
Comment (zhuangchong) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/exception/BaseException.java:23]: Used to handle underlying exceptions and resolve SNOAR disallowed used 'Exception'

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**